modality-probe-debug-collector/src/lib.rs

```diff
@@ -0,0 +1,1142 @@
+use chrono::Utc;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fs::File;
+use std::net::SocketAddrV4;
+use std::path::PathBuf;
+use std::rc::Rc;
+use std::sync::mpsc::Receiver;
+use std::thread::sleep;
+use std::time::Duration;
+
+use probe_rs::{MemoryInterface, Session};
+
+use modality_probe::compact_log::CompactLogItem;
+use modality_probe::{EventId, LogicalClock, ProbeId};
+use modality_probe_udp_collector::add_log_report_to_entries;
+use race_buffer::reader::{RaceBufferReader, Snapper};
+use race_buffer::Entry;
+use util::alloc_log_report::LogReport;
+use util::model::{LogEntry, SegmentId, SessionId};
+
+type Result<T> = std::result::Result<T, Box<dyn Error>>;
+
+// NOTE: These may be changed once RaceBuffer is implemented into ekt
+const PROBE_ID_OFFSET: u32 = 0x0;
+const BUF_ADDR_OFFSET: u32 = 0x4;
+const BUF_CAP_OFFSET: u32 = 0x8;
+const WCURS_OFFSET: u32 = 0xc;
+//const _EVENT_COUNT_OFFSET: u32 = 0x0;
+//const _CLOCKS_ADDR_OFFSET: u32 = 0x0;
+//const _CLOCKS_LEN_OFFSET: u32 = 0x0;
```

Comment:
Are these commented out lines still relevant?

Reply:
I wasn't sure if they would be needed later, but I'll delete them for now.